src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -436,6 +436,8 @@ public string ToString(int startIndex, int length)
 
         public StringBuilder Clear()
         {
+            // Retain enough storage to store the same size content, plus 20% buffer, to help avoid reallocating during repeated use
+            Capacity = Math.Max(DefaultCapacity, Math.Min(Capacity, (int)(Length * 1.2)));
```

Comment:
I would consider Clear() to be on the 'hot' path, so I care about it being 'lean'   My concern her is that setting the capacity here forces it to potentially copy the data that is currently in the stringbuilder to its new home THEN clear it (which is clearly a waste)

This also does not fix the issue if someone uses sb.Length=0 (instead of Clear()). 

Thus I would prefer if the change is in Length
```
                    if (chunk != this)
                    {
                        // Avoid possible infinite capacity growth.  See https://github.com/dotnet/coreclr/pull/16926
                        int capacityToPreserve = Math.Min(Capacity, Math.Max(Length * 5 / 4, DefaultCapacity));

                        // We crossed a chunk boundary when reducing the Length. We must replace this middle-chunk with a new larger chunk,
                        // to ensure the original capacity is preserved.
                        int newLen = capacityToPreserve - chunk.m_ChunkOffset;
```
While you are in there I would also suggest putting an #if DEBUG around the call to 
*                 int originalCapacity = Capacity;
Since this is on the critical path for Clear() and originalCapacity is only (now) used for Debug code and I am not sure that the JIT optimizer could remove the code associated with this statement completely.

Reply:
I tweeked the code above to change DefaultCapacity to m_ChunkChars.Length.    I did this because it is best to preserve (as much as we can) the invariant that Capacity does not shrink (to preserve the Stringbuilder growing until a high water mark and then being efficient thereafter).